### DISCUSSION TITLE
Feedback for creating new power sensing hardware

### DISCUSSION THREAD

**Discussion by ajlennon** [arendst/Tasmota#15380] [Ideas]
Hi everybody,

We've been working with Tasmota for some time now and we love it. Real proper decent embedded engineering.

So firstly thanks for all the hard work everybody has put into making this awesome project.

We've been having lots and lots of problems sourcing power sensing plugs from China that we can successfully upgrade (the Tuya internal IC and firmware lockdown issues I think). 

So we've come to the conclusion we might design our own plug and get it certified.

I would like this plug to be open and updateable to any version of Tasmota (or other firmware such as ESPurna) that users want to put on the hardware they purchase.

I also don't particularly want to recreate the wheel so I am asking if there are others who already have such hardware projects underway, and if not if people would like to feed back their needs and wants for such a device here to see what we might do?

Thanks!

Alex Lennon

**Comment by arendst** [comment]
You might want to look here https://templates.blakadder.com/preflashed.html first.

**Reply by ajlennon** [reply]
> You might want to look here https://templates.blakadder.com/preflashed.html first.

This is really interesting. Have reached out to them. Thanks !!!

**Comment by barbudor** [comment]
Hi
If you want to design you own plug, I would advise having a look to my prefered one : Gosund SP1 or now Nous A1:
- rated 16A 🤔
- compact : you can have multiple side by side on a power strip without issue
- easy to open by the screw in the French-earth hole
- easy to serial flash as the required gpios are easily reachable (a simple hand made jig avoid dismantling the PCB)
- nice little antenna, probably better than pcb antenna

**Reply by ajlennon** [reply]
Awesome thanks - I have been using Avatar, 2NICE and Gosund. That all sounds really rather cool. Thanks for the advice!

**Reply by joba-1** [reply]
> my prefered one : Gosund SP1 or now Nous A1

mine as well. Will be very hard to beat :)

**Comment by ajlennon** [comment]
Thanks for all the advice everybody. And thanks for the pointer @arendst. I've had a chat with the folks over at ATHOM and have a couple of their plugs on order! Brill. Thanks again
